druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/RocksDBLookupExtractor.java

```diff
@@ -41,6 +56,70 @@ public boolean canIterate() {
 
     @Override
     public Iterable<Map.Entry<String, String>> iterable() {
-        return staticMap.entrySet();
+        if (rocksDBManager == null) {
+            LOG.info("rocksDBManager is null!!");
+        }
+        else {
+            LOG.info("rocksDBManager: " + rocksDBManager);
+        }
+        if (extractionNamespace == null) {
+            LOG.info("extractionNamespace is null!!");
+        }
+        else {
+            LOG.info("extractionNamespace: " + extractionNamespace);
+        }
+        if (extractionNamespace.getNamespace() == null) {
+            LOG.info("extractionNamespace.getNamespace() is null!!");
+        }
+        else {
+            LOG.info("extractionNamespace.getNamespace(): " + extractionNamespace.getNamespace());
+        }
+        final RocksDB db = rocksDBManager.getDB(extractionNamespace.getNamespace());
+        LOG.info("Got RocksDB db");
+        if (db == null) {
+            LOG.info("RocksDB db is null!!");
+        }
+        else {
+            LOG.info("RocksDB db: " + db.toString());
+        }
+
+        Map<String, String> tempMap = new java.util.HashMap<>();
+        RocksIterator it = db.newIterator();
+        LOG.info("Got RocksIterator it");
+        it.seekToFirst();
+        LOG.info("Got seekToFirst");
+        while (it.isValid()) {
+            LOG.info("Iterator is valid");
+            LOG.info("it.key: " + Arrays.toString(it.key()));
+            LOG.info("it.value: " + Arrays.toString(it.value()));
+            StringBuilder sb = new StringBuilder();
+            try {
+                byte[] cacheByteValue = db.get(it.key());
+                LOG.info("Got cacheByteValue");
+                LOG.info("cacheByteValue: " + Arrays.toString(cacheByteValue));
+                Parser<Message> parser = schemaFactory.getProtobufParser(extractionNamespace.getNamespace());
```

Comment:
it can be protobuf  as well as flatbuffers.  can you split DerSer into two methods?

Reply:
Different classes handle the different serializations. As discussed, we can have flatbuffer SerDe in the corresponding RocksDBLookupExtractorFlatBuffer class.